packages/common/log/src/processors/file-processor.ts

```diff
@@ -52,7 +54,31 @@ export const createFileProcessor = ({
       },
       context: jsonify(getContextFromEntry(entry)),
     };
-    appendFileSync(fd, JSON.stringify(record) + '\n');
+    let retryTS: number = 0;
+
+    // Retry writing if EAGAIN is encountered.
+    //
+    // Node may set stdout and stderr to non-blocking. https://github.com/nodejs/node/issues/42826
+    // This can cause EAGAIN errors when writing to them.
+    // In order to not drop logs, make log methods asynchronous, or deal with buffering/delayed writes, spin until write succeeds.
+
+    while (true) {
+      try {
+        return appendFileSync(fd, JSON.stringify(record) + '\n');
+      } catch (err: any) {
+        if (err.code !== 'EAGAIN') {
+          throw err;
+        }
+        if (retryTS === 0) {
+          retryTS = performance.now();
+        } else {
+          if (performance.now() - retryTS > EAGAIN_MAX_DURATION) {
```

Comment:
It's concerning that this can block the thread for up to 1 second. Not sure if it ever is gonna happen in practice

Reply:
yeah, it's definitely a corner case, a very large log message that exceeds the buffer size of the stderr pipe (which I'm checking whether we have any control over increasing), but it's also something this should be able to handle safely and predictably.

1s was a super high defensive estimate, I don't think it should ever be more than a ms or two in practice.